src/dispatch_private.h

```diff
@@ -141,4 +141,16 @@ void qd_dispatch_unregister_entity(qd_dispatch_t *qd, void *impl);
 /** Set the agent */
 void qd_dispatch_set_agent(qd_dispatch_t *qd, void *agent);
 
+/**
```

Comment:
Yeah, I tried it that way also.  The advantage of this is that I can pass in NULL to cause the memory to be freed.  But, the copy in may be the better way.  Only issue, is if you look at the two places this is being use, one is passing strdup, so no problem there, the strdup could just happen inside the method.  The other one is passing the result of qd_entity_opt_string().  That also returns newly allocated memory, right?  So, I'd have to catch that return separately, pass it in and then free it.  And, we'd basically be doing a wasted extra alloc/free.  That didn't seem so nice.  So, that's why I tended toward having this internal method just manage the setting of the pointer and ensuring that prior pointee's are freed.  And, of course, that's why I  commented the method as taking ownership of the memory.  Let me know if you prefer the other approach.